shell.nix

```diff
@@ -0,0 +1,107 @@
+# This shell.nix file is designed for use with cabal build
+# It aims to leverage the nix cache in as much as possible
+# while reducing Nix maintenance costs.
+# It does **not** aim to replace Cabal/Stack with Nix
+
+# Maintaining this file:
+#
+# - Dealing with broken nix-shell
+#
+#     1. Bump the nixpkgs version using `niv update nixpkgs`
+#     2. Comment out any remaining failing packages
+#
+# - Dealing with broken cabal build inside nix-shell:
+#
+#    If my understanding of cabal new-build is correct this should never happen,
+#    assuming that cabal new-build does succeed outside nix-shell
+
+{ sources ? import nix/sources.nix,
+  nixpkgs ? import sources.nixpkgs {},
+  compiler ? "default",
+  hoogle ? false
+ }:
+with nixpkgs;
+
+let defaultCompiler = "ghc" + lib.replaceStrings ["."] [""] haskellPackages.ghc.version;
+    haskellPackagesForProject = p:
```

Comment:
this bit was a bit confusing, but I see why. Otoh, why not let nix build the dependecies?

Reply:
We let Nix download the dependencies! We don't want Nix to build them, it's slow and often wrong, since it doesn't do constraint solving.